### PR TITLE
Optimize: convert trusted keys list to a set for better performance

### PR DESCRIPTION
**Pull request by LuizWT**
# PR:

This pull request optimizes the handling of trusted keys by converting lists to sets. This change improves performance for membership checks, making them faster.

# Tests

No new tests were added, as this is a refactor that does not change the business logic.

### CODE REVIEW

**Comment by laanwj** [comment]
Have you benchmarked the performance difference?

File: contrib/verify-commits/verify-commits.py [hunk 1]
```diff
@@ -83,17 +83,17 @@ def main():
     dirname = os.path.dirname(os.path.abspath(__file__))
     print("Using verify-commits data from " + dirname)
     with open(dirname + "/trusted-git-root", "r", encoding="utf8") as f:
-        verified_root = f.read().splitlines()[0]
+        verified_root = set(f.read().splitlines())[0]
```
Comment (laanwj) [inline contrib/verify-commits/verify-commits.py:86]: Have you even tested this? You can't subscript index into a set, and it makes no sense.

Comment (SanAndreyas) [reply contrib/verify-commits/verify-commits.py:86]: This is not even an equivalent expression. When you convert a list to a set, the order of the lines is lost. Therefore, accessing the first element with [0] could give you any of the lines randomly, not necessarily the first line from the original file. In this case you needed index 0 ( root ).

**Comment by l0rinc** [comment]
Approach NACK - no explanation, no benchmarks, no explanation for how to test this, etc

File: contrib/verify-commits/verify-commits.py (hunk 1 above)
Comment (LuizWT) [reply contrib/verify-commits/verify-commits.py:86]: > This is not even an equivalent expression. When you convert a list to a set, the order of the lines is lost. Therefore, accessing the first element with [0] could give you any of the lines randomly, not necessarily the first line from the original file. In this case you needed index 0 ( root ).

You're right. It makes no sense to use a set in this context. Thanks for pointing this out

**Pull request closed without merging**